### PR TITLE
Adding CORS Documentation

### PR DESCRIPTION
**Pull request by jrshaffe**
Adding some CORS related info to the api auto registration feature.

Which other branches should this be merged with (if any)? TAP 1.3

### CODE REVIEW

File: api-auto-registration/usage.hbs.md [hunk 1]
```diff
@@ -114,4 +114,14 @@ spec:
         kind: HTTPProxy
         name: my-httpproxy
 
-```
\ No newline at end of file
+```
+
+## <a id='cors'></a>CORS
```
Comment (shrutiyer) [inline api-auto-registration/usage.hbs.md:119]: Maybe a little more descriptive title would read better? Setting up CORS for OpenAPI specs

File: api-auto-registration/usage.hbs.md [hunk 2]
```diff
@@ -114,4 +114,14 @@ spec:
...
+
+The [CORS](https://fetch.spec.whatwg.org/#http-cors-protocol) protocol is used by the agent (usually a browser) to verify whether an API is used by the current origin. Your API must be configured to allow CORS Requests from TAP GUI. How you accomplish this varies based on the programming language and framework you are using. If you are using Spring you can get more information [here](https://spring.io/blog/2015/06/08/cors-support-in-spring-framework). If you are using something else please refer to it's documentation.
```
Comment (shrutiyer) [inline api-auto-registration/usage.hbs.md:121]: Maybe add a sentence or two at the beginning of this paragraph on who might encounter this error. Like "If you want to use the "Try it out" feature for OpenAPI specs from the API Docs plugin, you will need to setup CORS to allow successful requests" or something like that

File: api-auto-registration/usage.hbs.md [hunk 3]
```diff
@@ -114,4 +114,14 @@ spec:
...
+
+At a high level, the TAP GUI domain must be accepted as valid cross-origin by your API. Verify the following:
+
+- **Origins allowed** (header: `Access-Control-Allow-Origin`): a list of comma-separated values. This list must include your TAP GUI host.
```
Comment (shrutiyer) [inline api-auto-registration/usage.hbs.md:125]: Nice

File: api-auto-registration/usage.hbs.md (hunk 1 above)
Comment (jrshaffe) [reply api-auto-registration/usage.hbs.md]: Changed

File: api-auto-registration/usage.hbs.md (hunk 2 above)
Comment (jrshaffe) [reply api-auto-registration/usage.hbs.md]: changed

**Review by shrutiyer — APPROVED** [review]

**Comment by RichardJJG** [comment]
You want the `1-3-1` branch as well, I'm guessing, @jrshaffe?

**Comment by jrshaffe** [comment]
@RichardJJG yes, I would like it into the 1-3-1 branch as well.  Do I need to open up a PR for that?

**Comment by RichardJJG** [comment]
@jrshaffe, no, it's enough just to tell us that. A writer will cherry-pick after finishing editing the original PR content.

**Comment by jrshaffe** [comment]
@RichardJJG

**Pull request merged**